specs/interop/token-bridging.md

```diff
@@ -72,26 +101,200 @@ contract OptimismSuperchainERC20 is ERC20, ISuperchainERC20 {
 }
 ```
 
-## L1 Native Tokens
+[Disclaimer]
+
+- Threat the following code as experimental. It will have errors and will change function names. The goal is to discuss functionalities.
+- Current function names are open to discussion.
+
+## Implementation
+
+### Cross Chain transfers
+
+The standard will include the following three external functions:
+
+- `bridgeERC20(uint256 amount, uint256 chainId)`
+- `bridgeERC20To(address to, uint256 amount, uint256 chainId)`
+- `finalizeBridgeERC20(address to, uint256 amount)`
+
+The first two functions will burn `amount` tokens and initialize a message to the `L2ToL2CrossChainMessenger` to mint the `amount` in the destination (`to` address in `chainId`).
+
+`finalizeBridgeERC20(address to, uint256 amount)` will process incoming messages from the `L2ToL2CrossChainMessenger` initiated from the same token on a different `chainId` and mint `amount` to the `to` address.
+
+A possible implementation would look like this:
+
+```solidity
+function bridgeERC20(uint256 _amount, uint256 _chainId) external returns (bool success) {
+    success = bridgeERC20To(msg.sender, _amount, _chainId);
+  }
+
+function bridgeERC20To(address _to, uint256 _amount, uint256 _chainId) public returns (bool) {
+  _burn(msg.sender, _amount);
+  bytes memory _message = abi.encodeCall(this.finalizeBridgeERC20, (_to, _amount));
+  _sendMessage(_chainId, _message);
+  return true;
+}
+
+function finalizeBridgeERC20(address _to, uint256 _amount) external onlyMessenger {
+  _mint(_to, _amount);
+}
+
+/////// Internals and modifiers ///////////
+function _sendMessage(uint256 _destination, bytes memory _data) internal virtual {
+  L2ToL2CrossDomainMessenger.sendMessage(_destination, address(this), _data);
+}
+
+modifier onlyMessenger() virtual {
```

Comment:
For the spec, its easier to read the invariants when they are inlined rather than in a modifier, but the modifier is def easier to maintain